### DISCUSSION TITLE
Command : cast call

### DISCUSSION THREAD

**Discussion by Perso77** [Cyfrin/foundry-full-course-cu#46] [Q&A] [answered]
normally, the return of the retrieve() function must be 456 instead of 0 !. 
the same code works fine on remix
![call](https://github.com/Cyfrin/foundry-full-course-f23/assets/116321001/09cb6fe8-7e4a-4587-bc03-602b221488f4)
![code](https://github.com/Cyfrin/foundry-full-course-f23/assets/116321001/769754ea-5296-4bc9-bb69-0b64a495a612)

**Comment by cromewar** [comment]
Trying to reproduce this problem, any clue on this @PatrickAlphaC ?

**Reply by Perso77** [reply]
I think forge create and forge script script are two ways to deploy a contract. (please confirm if it is incorrect).

so, if i use forge script script, i have a problem with the call command (result: 0x)
forge create: works fine
@alymurtazamemon 
![call](https://github.com/Cyfrin/foundry-full-course-f23/assets/116321001/2288dd32-1c6b-483b-8bc5-3da7b5a41b65)

![call com](https://github.com/Cyfrin/foundry-full-course-f23/assets/116321001/de213f58-3898-4fc2-b212-003ea31f6bf5)

**Reply by alymurtazamemon** [reply, marked as answer]
@Perso77 `create` command deploys the contract and you can do the same with scripts but this will not affect the `cast send` command because it is for sending transactions. It just needs a deployed contract address, RPC URL and private key to sign the tx, it does not matter you use `create` or script to deploy the contract.

**Reply by cromewar** [reply]
You are amazing @alymurtazamemon !

**Reply by DukeLurkin** [reply]
I tried this plus the suggestion on another discussion for adding --broadcast but neither have worked.  I'm still just getting 0x as a return on the retrieve(). Does anyone have any suggestions?

**Reply by cromewar** [reply]
I was trying to reproduce this, but I cannot get it to return me the same response as you. have you tried cloning the repo @DukeLurkin?

**Comment by alymurtazamemon** [comment]
@Perso77 It will show but make sure your contract address is correct;
```bash
alymurtazamemon@Alis-MacBook-Air foundry-simple-storage % forge create SimpleStorage --rpc-url anvil --private-key $ANVIL_PRIVATE_KEY    
[⠆] Compiling...
No files changed, compilation skipped
Deployer: 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266
Deployed to: 0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0
Transaction hash: 0xb663053e044871f46c03f299c5d60f01a83e350057b48876403ab26fae9ea792
alymurtazamemon@Alis-MacBook-Air foundry-simple-storage % cast send 0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0 "store(uint256)" 456 --rpc-url anvil --private-key $ANVIL_PRIVATE_KEY


blockHash               0x8bd9ac924ac6959a0d47ef34afc6342a8687e55b455df9fbecdaca6424f1467a
blockNumber             4
contractAddress         
cumulativeGasUsed       43550
effectiveGasPrice       3674819918
gasUsed                 43550
logs                    []
logsBloom               0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
root                    
status                  1
transactionHash         0x848ab9343d7902896e72664b8d00b8d70f23781b060cff64da2a956968dc09b9
transactionIndex        0
type                    2
alymurtazamemon@Alis-MacBook-Air foundry-simple-storage % cast call 0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0 "retrieve()"
0x00000000000000000000000000000000000000000000000000000000000001c8

alymurtazamemon@Alis-MacBook-Air foundry-simple-storage % cast --to-base 0x00000000000000000000000000000000000000000000000000000000000001c8 dec   
456
alymurtazamemon@Alis-MacBook-Air foundry-simple-storage %    
```
